types/agent.go

```diff
@@ -83,10 +85,11 @@ type AgentsParams struct {
 
 // UpdateAgent request payload for updating an agent.
 type UpdateAgent struct {
-	Name      *string          `json:"name"`
-	Version   *string          `json:"version"`
-	Edition   *AgentEdition    `json:"edition"`
-	Flags     *[]string        `json:"flags"`
-	RawConfig *string          `json:"rawConfig"`
-	Metadata  *json.RawMessage `json:"metadata"`
+	Name          *string          `json:"name"`
+	Version       *string          `json:"version"`
+	Edition       *AgentEdition    `json:"edition"`
+	Flags         *[]string        `json:"flags"`
+	RawConfig     *string          `json:"rawConfig"`
+	Metadata      *json.RawMessage `json:"metadata"`
+	EnvironmentID *string          `json:"environmentID"`
```

Comment:
This allows for migrating an agent from one environmment to another.

Maybe is something that we perform in a different endpoint or in a different way, or we don't implement it at all.

Reply:
Not yet, lets dismiss this part.